### PR TITLE
Timestamp literal accepts a timezone param

### LINKED ISSUE #1712: `ibis.timestamp` should accept a timezone argument
`ibis.timestamp('2017-01-01', timezone='UTC')`, similar to the signature of `dt.Timestamp`

### PR DESCRIPTION
**Pull request by toryhaavik**
closes #1712

### CODE REVIEW

File: ibis/expr/tests/test_value_exprs.py [hunk 1]
```diff
@@ -1322,6 +1322,13 @@ def test_large_timestamp():
     assert result == expected
 
 
+def test_timestamp_with_timezone():
```
Comment (xmnlab) [inline ibis/expr/tests/test_value_exprs.py]: Maybe would be good adding one more test using timezone=None

File: ibis/expr/tests/test_value_exprs.py [hunk 2]
```diff
@@ -1322,6 +1322,17 @@ def test_large_timestamp():
     assert result == expected
 
 
+@pytest.mark.parametrize('tz', [
+    None,
+    'UTC'
+])
+def test_timestamp_with_timezone(tz):
```
Comment (cpcloud) [inline ibis/expr/tests/test_value_exprs.py]: Can you add a second that checks that the `timezone` is propagated to the `type()` of the resulting expression?

**Comment by cpcloud** [comment]
LGTM, merging.

**Comment by cpcloud** [comment]
Thanks @toryhaavik !
